core/internal/cltest/simulated_backend.go

```diff
@@ -330,7 +329,13 @@ func (c *SimulatedBackendClient) SubscribeNewHead(
 				default:
 					head := &models.Head{Number: h.Number.Int64(), Hash: h.Hash(), ParentHash: h.ParentHash, Parent: lastHead}
 					lastHead = head
-					channel <- head
+					select {
```

Comment:
Say you run a test using the geth sim, upon head tracker shutdown:
1. We break out of the head tracker loop which is reading heads
2. A geth sim mines a new block
3. We call unsubscribe which will hang because this goroutine is unable to read from ```channel```
4. It times out because of our timedUnsubscribe
5. We close ```channel``` then this ```channel<-head``` panics trying to send on a closed channel